cubeviz/tests/test_specviz_integration.py

```diff
@@ -0,0 +1,57 @@
+"""
+Contains tests of integration between specviz and cubeviz
+
+In general, most of these tests should be in the specviz->cubeviz direction. In
+other words, it should test whether events in specviz cause the appropriate
+response in cubeviz. Tests in the other direction are probably better off being
+added to existing cubeviz tests.
+"""
+
+import pytest
+from glue.core import roi
+
+
+@pytest.mark.parametrize('slice_index', [0, 100, 1000, 1024])
+def test_specviz_slice_update(cubeviz_layout, slice_index):
+    """
+    Make sure slice updates in specviz propagate to cubeviz.
+    """
+    specviz = cubeviz_layout.specviz._widget
+    wavelength = cubeviz_layout._slice_controller._wavelengths[slice_index]
+
+    specviz._slice_indicator.setPos([wavelength, 0])
+    assert cubeviz_layout._slice_controller.synced_index == slice_index
+
+
+def test_cubeviz_roi(cubeviz_layout):
+    """
+    Test whether ROI creation in cubeviz propagates to specviz
+    """
+    specviz = cubeviz_layout.specviz._widget
+    assert len(specviz._layer_artist_container.layers) == 1
+
+    # Create ROI/subset
+    cl_viewer = cubeviz_layout.split_views[1]._widget
+    cl_viewer.apply_roi(roi.CircularROI(xc=6, yc=10, radius=3))
+
+    assert len(specviz._layer_artist_container.layers) == 2
+
+    # Remove the ROI/subset
+    dc = cubeviz_layout.session.application.data_collection
+    dc.remove_subset_group(dc.subset_groups[0])
+
+    assert len(specviz._layer_artist_container.layers) == 1
+
+
+def tests_wavelength_units_update(cubeviz_layout):
+    """
+    Make sure wavelength unit changes in specviz are reflected in cubeviz.
+    """
+    specviz = cubeviz_layout.specviz._widget
+    current_units = str(cubeviz_layout._wavelength_controller.current_units)
+
+    specviz.hub.plot_widget.set_spectral_axis_unit('Angstrom')
```

Comment:
Small change, but you may want to use the [`update_units`](https://github.com/spacetelescope/specviz/blob/master/specviz/third_party/glue/viewer.py#L423) method in the data viewer instead of settings the unit explicitly on the plot widget (since CubeViz uses it this way). Eventually, I want to fix the wavelength update message and have the slice indicator updated in this method as well.

Reply:
@nmearl the reason I didn't do that is because it would require me to pass new values for both spectral units and flux units and I wanted to keep the test as isolated/direct as possible.

Reply:
@drdavella I'm not sure I follow. You can just call it like `specviz.update_units(spectral_axis_unit='Angstrom')`, the default value of `None` for flux will ensure that the flux units are left alone. Furthermore, the `set_spectral_axis_unit` will raise the `unit_changed` event (this is different from setting the `spectral_axis_unit` directly, which does not raise an event), which is connected back to cubeviz (it shouldn't _do_ anything, but you're basically doing a loop of `cubeviz unit changed -> specviz unit changed -> cubeviz unit changed`).

Reply:
@nmearl sorry, I think I must have been looking in the wrong place or at the wrong function. I'll make the update.

Reply:
It turns out I actually need to call these methods directly rather than the higher-level one. I need the events to occur since the purpose of the test is to determine whether unit change events in specviz propagate properly to cubeviz.